submissions/Shrai-dev/oop-exercise/index.js

```diff
@@ -0,0 +1,78 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: https://github.com/Shrai-dev/a-tiny-JS-world
+   Web app: https://shrai-dev.github.io/a-tiny-JS-world/
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Inhabitant {
+   constructor (species, name, gender, age, saying, friends) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.age = age;
+      this.saying = saying;
+      this.friends = friends;
+   }
+   toString() {
+      return `I am a ${this.species}. My name is ${this.name}. I am ${this.age} years old. My friends are ${this.friends.join(', ')}. I say: ${this.saying}`
+   }
+
+}
+
+class Animal extends Inhabitant {
+   constructor(species, name, gender, age, paws, saying, friends) {
+      super(species, name, gender, age, saying, friends);
+      this.paws = paws;
+   }
+   toString() {
+      return super.toString() + ` I have ${this.paws} paws.`
+   }
+}
+
+class Human extends Inhabitant {
+   constructor (species, name, gender, age, legs, hands, saying, friends) {
+      super(species, name, gender, age, saying, friends);
+      this.legs = legs;
+      this.hands = hands;
+   }
+   toString() {
+      return super.toString() + ` I have ${this.legs} legs and ${this.hands} hands.`
+   }
+
+}
+
+// ======== OUTPUT ========
+
+const dog = new Animal ('dog', 'Buddy', 'male', 3, 4, 'bark-bark!', ['Elsa', 'Chip', 'Max']);
+const cat = new Animal ('cat', 'Chloe', 'female', 2, 4, 'meow!', ['Leo', 'Simon', 'Kitty']);
+const man = new Human ('human', 'Brian', 'male', 26, 2, 2, 'Everyone thinks of changing the world, but no one thinks of changing himself!', ['Ethan', 'James', 'Ryan']);
+const woman = new Human ('human', 'Olivia', 'female', 24, 2, 2, 'The simplest way to be happy is to do good!', ['Maria', 'Kelly', 'Jennifer']);
```

Comment:
Sending a species string value to a species class looks like unnecessary duplication.
Well, `Human` is a species. `Animal` is not. If you had `Dog` and `Cat` classes you could avoid this for them too.

Reply:
@OleksiyRudenko Thanks for pointing to that. I've fixed these duplications.